warrior/WarriorCore/Classes/html_results_class.py

```diff
@@ -9,432 +9,172 @@
-        return href_link
+    def create_line_result(self, line, type):
+        """ create new objs"""
+        temp = LineResult()
+        temp.set_html(line, type, self.steps)
+        self.lineObjs.append(temp)
+        self.lineCount += 1
 
-    def _get_html_resultspath(self):
-        """Get the results path for the 
-        html results file"""
+    def set_line_objs(self):
+        """ call to create a new obj per item"""
+        self.lineCount = 0
+        project_node_list = [self.junit_root]
+        for project_node in project_node_list:
+            self.create_line_result(project_node, "Project")
+            for testsuite_node in project_node.findall("testsuite"):
+                self.create_line_result(testsuite_node, "Testsuite")
+                for testcase_node in testsuite_node.findall("testcase"):
+                    self.create_line_result(testcase_node, "Testcase")
+                    self.steps = 0
+                    for step_node in testcase_node.findall("properties"):
+                        for node in step_node.findall("property"):
+                            if node.get('type') == 'keyword':
+                                self.steps += 1
+                                self.create_line_result(node, "Keyword")
+
+    def get_path(self):
+        """ get the html results path """
         filename = file_Utils.getNameOnly(os.path.basename(self.junit_file))
         filename = filename.split("_junit")[0]
         html_filename = filename + ".html"
-        results_dir = os.path.dirname(self.junit_file)
-        html_results_path = results_dir + os.sep + html_filename
+        if hasattr(self, 'givenPath'):
+            html_results_path = self.givenPath + os.sep + html_filename
+        else:
+            results_dir = os.path.dirname(self.junit_file)
+            html_results_path = results_dir + os.sep + html_filename
+
         return html_results_path
 
+    def merge_html(self, dynamic_html):
+        """ merge html from template and dynamic """
+        temp = open(self.html_template)
+        templateHTML = temp.read().replace('\n', '')
+        temp.close()
+        index = templateHTML.rfind('</table>')
+        return templateHTML[:index] + dynamic_html + templateHTML[index:] + self.get_war_version()
+
+    def get_war_version(self):
+        """ find the warrior version """
+        path = self.get_path().split('warriorframework')[0] + 'warriorframework/version.txt'
+        if os.path.isfile(path):
+            version = open(path, 'r').read().split(':')[2]
+            return '<div class="version">' + version + '</div>'
+        else:
+            return ''
+
+    def generateHTML(self, junitObj, givenPath):
+        """ build the html """
+        if junitObj:
+            self.junit_file = junitObj
+            self.junit_root = xml_Utils.getRoot(self.junit_file)
+        if givenPath:
+            self.givenPath = givenPath
+
```

Comment:
generateHTML() will never enter the if block on line 159 - since the only place generateHTML() is being called is on line 226 of this file  and the value for givenPath is being set to None. Is there a reason why this block of code exists?